source/extensions/filters/http/jwt_authn/filter.cc

```diff
@@ -56,12 +65,35 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
     return Http::FilterHeadersStatus::Continue;
   }
 
-  // Verify the JWT token, onComplete() will be called when completed.
-  const auto* verifier =
-      config_->findVerifier(headers, *decoder_callbacks_->streamInfo().filterState());
-  if (!verifier) {
+  const Verifier* verifier = nullptr;
+  bool use_per_route = false;
+  Router::RouteConstSharedPtr route = decoder_callbacks_->route();
+  if (route != nullptr && route->routeEntry() != nullptr) {
```

Comment:
You don't need this `if`, `resolveMostSpecificPerFilterConfig` takes care of this.

Reply:
Good point